collections/nemo_nlp/nemo_nlp/data/datasets/utils.py

```diff
@@ -547,6 +547,131 @@ def merge(data_dir, subdirs, dataset_name, modes=['train', 'test']):
     return outfold, none_slot
 
 
+def get_intent_query_files_dialogflow(path):
+    fileslist = []
+    for root, _, files in os.walk(path):
+        for file in files:
+            if '_usersays_en.json' in file:
+                fileslist.append(os.path.join(root, file))
+    return fileslist
+
+
+def get_intents_slots_dialogflow(files, slot_labels):
+
+    intent_names = []
+    intent_queries = []
+    slot_tags = []
+
+    for index, file in enumerate(files):
+        intent_names.append(os.path.basename(file).split('_usersays')[0])
+
+        with open(file) as json_file:
+            intent_data = json.load(json_file)
+            for query in intent_data:
+                querytext = ""
+                slots = ""
+                for segment in query['data']:
+                    querytext = ''.join([querytext, segment['text']])
+                    if 'alias' in segment:
+                        for word in segment['text'].split():
+                            slots = ' '.join([
+                                    slots,
+                                    slot_labels.get(segment['alias'])])
```

Comment:
join might be an overkill here. just use:
slots + ' ' + slot_labels[segment['alias']]

get() method is useful when you use it with a default value (if the key doesn't exist, use this value). if you don't use it, then just use []